src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -187,6 +187,14 @@ export default function ConfirmSwapModal({
     doesTradeDiffer: Boolean(doesTradeDiffer),
   })
 
+  const swapFailed = Boolean(swapError) && !didUserReject(swapError)
+  useEffect(() => {
+    // Reset the modal state if the user rejected the swap.
+    if (swapError && !swapFailed) {
+      onCancel()
+    }
+  }, [onCancel, swapError, swapFailed])
```

Comment:
we need to use `useEffect` here to reset the ConfirmModalState since the state is local and the rejection happens in `useUniversalRouterSwapCallback`. maybe a future improvement could be made here if we start using a Context to manage Swap state that all children can access equally?